aiida/restapi/translator/node.py

```diff
@@ -587,9 +587,10 @@ def get_node_shape(ntype):
             nodetype = main_node.node_type
             nodelabel = main_node.label
             display_type = nodetype.split('.')[-2]
-            description = main_node.get_description()
-            if description == '':
-                description = main_node.node_type.split('.')[-2]
+            try:
+                description = main_node.get_description()
+            except ValueError:
+                description = display_type
```

Comment:
there seems to be a few places where this is duplicated.
can this be encapsulated in one function that is used in these three places?

Reply:
done!

Reply:
I was checking how to move also the rest of the code inside a function and I noted that in 
```
            nodes.append({
                "id": node_count,
                "nodeid": pk,
                "nodeuuid": uuid,
                "nodetype": nodetype,
                "nodelabel": nodelabel,
                "displaytype": display_type,
                "group": "main_node",
                "description": description,
                "shape": get_node_shape(nodetype)
            })
```

not only  `group` and `node_count` values differ, but also in the 'duplicated' chunks there are two additional fields `linklabel` and `linktype`. Do you think that it still makes sense to put everything into a function?

Reply:
it's fine like this